addons/im_livechat/models/chatbot_script.py

```diff
@@ -182,19 +183,18 @@ def action_view_livechat_channels(self):
     # Tooling / Misc
     # --------------------------
 
-    def _format_for_frontend(self):
-        """ Small utility method that formats the script into a dict usable by the frontend code. """
-        self.ensure_one()
-
-        return {
-            'id': self.id,
-            'name': self.title,
-            'partner': {'id': self.operator_partner_id.id, 'type': 'partner', 'name': self.operator_partner_id.name},
-            'welcomeSteps': [
-                step._format_for_frontend()
-                for step in self._get_welcome_steps()
-            ]
-        }
+    def _to_store(self, store: Store, /, *, fields=None, **kwargs):
+        if fields is None:
+            fields = ["title", "operator_partner_id", "welcome_steps"]
+        for script in self:
+            data = script._read_format(
+                [f for f in fields if f not in {"operator_partner_id", "welcome_steps"}], load=False
+            )[0]
+            if "operator_partner_id" in fields:
+                data["operator_partner_id"] = Store.one(script.operator_partner_id, fields=["name"])
+            if "welcome_steps" in fields:
+                data["welcomeSteps"] = Store.many(script._get_welcome_steps())
```

Comment:
While we're at it, I would rename `welcomeSteps` to `welcome_steps` to maintain snake_case for Python fields. This facilitates record-to-store transformation and helps keep the field's origin clear.

Reply:
But, I don't think that `welcome_steps` is a python field. Rather it's a client side field. So, IMO we should keep it in `camelCase` itself.

Reply:
Oh yes sorry I thought it was a python field but you are right.

Reply:
I'm actually gravitating towards the opposite: if the field somehow ends up referenced in python (as it is the case here), then it becomes de facto a python field (not on a python model directly, but still interesting enough to be of python knowledge and to follow its guideline).

Reply:
Ok. I have renamed it to `welcome_steps`. But, I can't find it's use cases on client-side code. @tsm-odoo do you have any idea?

Reply:
It could be removed from the chat bot script model. But then, you have to update `get_session` to pass an entire script step, not the id.

Reply:
Then we can remove this in another PR where we remove the `_format_for_frontend` from `chatbot.script.step`.